lib/active_merchant/billing/gateways/borgun.rb

```diff
@@ -105,14 +105,14 @@ def add_3ds_fields(post, options)
 
       def add_3ds_preauth_fields(post, options)
         post[:SaleDescription] = options[:sale_description] || ''
-        post[:MerchantReturnURL] = options[:merchant_return_url] if options[:merchant_return_url]
+        post[:MerchantReturnURL] = options[:redirect_url] if options[:redirect_url]
       end
 
       def add_invoice(post, money, options)
         post[:TrAmount] = amount(money)
         post[:TrCurrency] = CURRENCY_CODES[options[:currency] || currency(money)]
         # The ISK currency must have a currency exponent of 2 on the 3DS request but not on the auth request
-        if post[:TrCurrency] == '352' && options[:apply_3d_secure] == '1'
+        if options[:apply_3d_secure] == '1'
           post[:TrCurrencyExponent] = 2
         else
           post[:TrCurrencyExponent] = 0
```

Comment:
Hmmmm should we flip the logic on this to be
```suggestion
        if options[:apply_3d_secure] != '1' && post[:TrCurrency] == '352'
          post[:TrCurrencyExponent] = 0
        else
          post[:TrCurrencyExponent] = 2
```

Reply:
Yea, this looks cleaner, except the `TrCurrencyExponent` should be `2` for all of the currencies (according to Borgun).